### PR TITLE
Add Collapsed Review View

### LINKED ISSUE #205: Review Card: Collapsed
Child issue of #199

**Designs**: [inVision Project](https://automattic.invisionapp.com/share/9RMY8JIJAVG#/screens/305602742)

screen shot 2018-07-13 at 1 01 56 pm

This issue depends on many of the components outlined in #199. Once those are in place, this task will involve bringing them all together and implementing the collapsed review card view.

### PR DESCRIPTION
**Pull request by emdashcodes**
This PR adds the collapsed card view and loading state view for reviews. It is not hooked up to an API yet (review endpoint has not been merged), so it uses the same loading trick as the Inbox tab. The actions also do nothing yet.

It also introduces a new package `interpolate-components` (https://github.com/Automattic/interpolate-components) that allows us to use components in translated strings.

Closes #205.

Mobile:

screen shot 2018-07-26 at 2 48 19 pm

screen shot 2018-07-26 at 2 48 03 pm

Desktop:

screen shot 2018-07-26 at 2 48 48 pm

screen shot 2018-07-26 at 2 48 27 pm

To Test:
* Load up the reviews panel and verify everything looks OK.

### CODE REVIEW

File: client/layout/activity-panel/panels/reviews.js [hunk 1]
```diff
@@ -2,101 +2,167 @@
...
 
 /**
  * Internal dependencies
  */
+import { ActivityCard, ActivityCardPlaceholder } from '../activity-card';
 import ActivityHeader from '../activity-header';
+import Gravatar from 'components/gravatar';
+import Link from 'components/link';
+import ProductImage from 'components/product-image';
+import { ReviewRating } from 'components/rating';
```
Comment (ryelle) [inline client/layout/activity-panel/panels/reviews.js]: We should probably standardize on whether a component exports itself or an object by default - maybe this is a container vs presentational component distinction, and we should split apart the components folder… 🤔 

(just pondering, not an actionable review comment)

**Review by ryelle — APPROVED** [review]
Thanks for adding `interpolateComponents`, I was wondering how we were going to do that :)

This looks good & is working well for me 👍

File: client/layout/activity-panel/panels/reviews.js (hunk 1 above)
Comment (justinshreve) [reply client/layout/activity-panel/panels/reviews.js]: In this case, the reason this is different is `components/rating` offers 3 different but related components. A base `Rating` component, and then a `ReviewRating` and `ProductRating` which provide wrappers around Rating, allowing you to just pass in a review and product object respectively.

File: client/layout/activity-panel/panels/reviews.js [hunk 2]
```diff
@@ -2,101 +2,167 @@
...
+	}
+
+	componentWillUnmount() {
+		clearTimeout( this.interval );
+	}
+
+	renderReview( review ) {
+		const product = demoProducts[ 0 ];
+
+		const title = interpolateComponents( {
```
Comment (timmyc) [inline client/layout/activity-panel/panels/reviews.js]: Cool!

**Review by timmyc — APPROVED** [review]
Looks great, and testing out well!

**Comment by justinshreve** [comment]
@woocommerce/woo-core-design Can I get a 👍 on if this is good to go?

**Comment by timmyc** [comment]
FYI I think they are all afk today @justinshreve

**Comment by timmyc** [comment]
FWIW sometimes i experience some random gravatar image loading shenanigans when my proxy is having problems... that could likely be the culprit here too.

**Pull request merged**